.circleci/verbatim-sources/linux-build-defaults.yml

```diff
@@ -37,11 +37,18 @@ pytorch_linux_build_defaults: &pytorch_linux_build_defaults
         export COMMAND='((echo "export BUILD_ENVIRONMENT=${BUILD_ENVIRONMENT}" && echo '"$NAMED_FLAG"' && echo "source ./workspace/env" && echo "sudo chown -R jenkins workspace && cd workspace && .jenkins/pytorch/build.sh") | docker exec -u jenkins -i "$id" bash) 2>&1'
         echo ${COMMAND} > ./command.sh && unbuffer bash ./command.sh | ts
 
-        echo $COMMAND
-
         # Push intermediate Docker image for next phase to use
         if [ -z "${BUILD_ONLY}" ]; then
-          export COMMIT_DOCKER_IMAGE=${DOCKER_IMAGE}-${CIRCLE_SHA1}
+          # Note [namedtensor build image]
+          # The namedtensor build uses the same docker image as
+          # pytorch-linux-trusty-py3.6-gcc5.4-build. In the push step, we have to
+          # distinguish between these two so the test can pick up the correct image.
+          output_image=${DOCKER_IMAGE}-${CIRCLE_SHA1}
+          if [[ ${BUILD_ENVIRONMENT} == *"namedtensor"* ]]; then
+            export COMMIT_DOCKER_IMAGE=$output_image-namedtensor
```

Comment:
Huh, I thought we had something that let us do this already, but I guess not.

Reply:
Nope, doesn't exist. So the XLA build is slightly broken in the respect that it's possible for xla-test to pull a non-xla build. However this doesn't happen in practice because the xla build takes a long time